src/components/EventForm/EventForm.tsx

```diff
@@ -0,0 +1,271 @@
+import { ChangeEvent } from "react";
+import Button from "../Button/Button";
+import InputBox from "../InputBox/InputBox";
+import { addDoc, collection, Timestamp } from "firebase/firestore";
+import { db } from "../../firebase";
+import { useState } from "react";
+import "./EventForm.scss";
+import { useNavigate } from "react-router-dom";
+import { isBefore } from "date-fns";
+
+type DefaultForm = {
+  eventName: string;
+  eventCategory: string;
+  eventCapacity: number | null;
+  eventDescription: string;
+  eventImages: string[];
+  eventDate: Timestamp | null;
+};
+
+type EventFormProps = {
+  handleNewEvent: () => void;
+};
+
+const EventForm = ({ handleNewEvent }: EventFormProps) => {
+  const [databaseInput, setDatabaseInput] = useState<DefaultForm>({
+    eventName: "",
+    eventCategory: "",
+    eventCapacity: null,
+    eventDescription: "",
+    eventDate: null,
+    eventImages: [""],
+  });
+  const [images, setImages] = useState<string[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [required, setRequired] = useState({
+    errorOne: true,
+    errorTwo: true,
+    errorThree: true,
+    errorFour: true,
+    errorFive: true,
+    errorSix: false,
+  });
+  const [submitted, setSubmitted] = useState<boolean>(false);
+
+  const handleEventName = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventName = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventName: eventName });
+    setRequired({ ...required, errorOne: databaseInput.eventName.length < 1 });
+  };
+  const handleEventCategory = (event: ChangeEvent<HTMLInputElement>) => {
+    const eventCategory = event.currentTarget.value;
+    setDatabaseInput({ ...databaseInput, eventCategory: eventCategory });
+    setRequired({
+      ...required,
+      errorTwo: databaseInput.eventCategory.length < 1,
+    });
+  };
+  const handleEventDate = (event: ChangeEvent<HTMLInputElement>) => {
+    const inputDate = new Date(event.currentTarget.value);
+    const currentDate = new Date();
+    const timestampDate = Timestamp.fromDate(inputDate);
+    setDatabaseInput({
```

Comment:
I think you will want to store this in a variable and check it before setting it to state.

```tsx
const isValidDate = isBefore(inputDate, currentDate)

if (isValidDate) {
  setDatabaseInput({...databaseInput, eventDate: timestampDate,});
}

  setRequired({ ...required, errorThree: isValidDate})
}